### PR TITLE
Setting gamemode as default now requires dbranks permission

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

If you change the gamemode the pop up asking you if you want to set the gamemode as default no longer appears if you don't have dbranks.
This time tested and working.

## Why It's Good For The Game

Admins won't change default gamemode anymore by accident.

## Changelog
:cl:
admin: Setting default gamemode now requires dbranks.
/:cl:

### CODE REVIEW

**Review by PowerfulBacon — COMMENTED** [review]
Use check_rights(R_DBRANKS) since it automatically checks usr.client

**Comment by Archanial** [comment]
> 
> 
> Use check_rights(R_DBRANKS) since it automatically checks usr.client

It also throws out "You don't have DBRANKS" everytime you change the gamemode.

**Comment by PowerfulBacon** [comment]
There is a parameter to make it not print that

File: code/modules/admin/topic.dm [hunk 1]
```diff
@@ -876,17 +876,21 @@
 			return
 
 		if (SSticker.HasRoundStarted())
-			if (askuser(usr, "The game has already started. Would you like to save this as the default mode effective next round?", "Save mode", "Yes", "Cancel", Timeout = null) == 1)
-				SSticker.save_mode(href_list["c_mode2"])
+			if(check_rights(R_DBRANKS, 0))
```
Comment (francinum) [inline code/modules/admin/topic.dm]:
```suggestion
			if(check_rights(R_DBRANKS, FALSE))
```

File: code/modules/admin/topic.dm [hunk 2]
```diff
@@ -876,17 +876,21 @@
...
+				if(askuser(usr, "The game has already started. Would you like to save this as the default mode effective next round?", "Save mode", "Yes", "Cancel", Timeout = null) == 1)
```
Comment (francinum) [inline code/modules/admin/topic.dm:880]:
```suggestion
				if(askuser(usr, "The game has already started. Would you like to save this as the default mode effective next round?", "Save mode", "Yes", "Cancel", Timeout = null) == TRUE)
```

File: code/modules/admin/topic.dm [hunk 3]
```diff
@@ -876,17 +876,21 @@
...
 			return
 		GLOB.master_mode = href_list["c_mode2"]
 		log_admin("[key_name(usr)] set the mode as [GLOB.master_mode].")
 		message_admins("<span class='adminnotice'>[key_name_admin(usr)] set the mode as [GLOB.master_mode].</span>")
 		to_chat(world, "<span class='adminnotice'><b>The mode is now: [GLOB.master_mode]</b></span>")
 		Game() // updates the main game menu
-		if (askuser(usr, "Would you like to save this as the default mode for the server?", "Save mode", "Yes", "No", Timeout = null) == 1)
-			SSticker.save_mode(GLOB.master_mode)
+		if(check_rights(R_DBRANKS, 0))
+			if(askuser(usr, "Would you like to save this as the default mode for the server?", "Save mode", "Yes", "No", Timeout = null) == 1)
```
Comment (francinum) [inline code/modules/admin/topic.dm]:
```suggestion
		if(check_rights(R_DBRANKS, FALSE))
			if(askuser(usr, "Would you like to save this as the default mode for the server?", "Save mode", "Yes", "No", Timeout = null) == TRUE)
```

**Comment by Isy232** [comment]
Yes please do this, I constantly fear pressing that button by accident when doing events...

File: code/modules/admin/topic.dm [hunk 4]
```diff
@@ -876,17 +876,21 @@
 			return
 
 		if (SSticker.HasRoundStarted())
-			if (askuser(usr, "The game has already started. Would you like to save this as the default mode effective next round?", "Save mode", "Yes", "Cancel", Timeout = null) == 1)
-				SSticker.save_mode(href_list["c_mode2"])
+			if(check_rights(R_DBRANKS, FALSE))
+				if(askuser(usr, "The game has already started. Would you like to save this as the default mode effective next round?", "Save mode", "Yes", "Cancel", Timeout = null) == TRUE)
```
Comment (PowerfulBacon) [inline code/modules/admin/topic.dm]: Does this not return the option they pressed, IE "Yes" or "Cancel"?

Comment (Archanial) [reply code/modules/admin/topic.dm]: I have no idea why but it returns 1 for yes and 2 for cancel

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**